src/read.c

```diff
@@ -4670,25 +4688,29 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
                         // One of the following invalid cases:
                         // * Multiple items are claiming to be the alpha auxiliary of the current item.
                         // * Alpha auxiliary is dimg for another item.
-                        avifFree(alphaItemIndices);
+                        avifFree(dimgIdxToAlphaItemIdx);
                         *alphaItem = NULL;
                         *isAlphaItemInInput = AVIF_FALSE;
                         return AVIF_RESULT_INVALID_IMAGE_GRID;
                     }
-                    alphaItemIndices[alphaItemCount++] = j;
+                    if (item->dimgIdx >= tileCount || dimgIdxToAlphaItemIdx[item->dimgIdx] != itemIndexNotSet) {
+                        avifFree(dimgIdxToAlphaItemIdx);
+                        *isAlphaItemInInput = AVIF_FALSE;
+                        return AVIF_RESULT_BMFF_PARSE_FAILED;
+                    }
+                    dimgIdxToAlphaItemIdx[item->dimgIdx] = j;
```

Comment:
As in the original code, I suggest we declare an `alphaItemCount` counter and increment it here. Then, after line 4713, we can check if `alphaItemCount != tileCount` and fail early. This allows us to delete the if statement at line 4743.